pkg/generate/app/sourcelookup.go

```diff
@@ -378,7 +380,7 @@ func StrategyAndSourceForRepository(repo *SourceRepository, image *ImageRef) (*B
 		Binary: repo.binary,
 	}
 
-	if repo.Info() != nil && repo.Info().Dockerfile != nil {
+	if (repo.ignoreRepository || repo.dockerfile) && repo.Info() != nil && repo.Info().Dockerfile != nil {
```

Comment:
`ignoreRepository` was set to true when creating a reference to a dockerfile as a single source for a build, so that this branch would be executed, while the next one would not (look at the whole function body)